fh-sync.d.ts

```diff
@@ -0,0 +1,412 @@
+// Type definitions for fh-sync
+// Project: https://github.com/feedhenry/fh-sync
+// Maintainer feedhenry-dev@redhat.com
+
+declare module SyncCloud {
+
+      /**
+       * Backoff strategies that can be passed to workers
+       */
+      type WorkerBackoffStrategy = 'fib'|'exp'
+
+      /**
+       * Valid actions (the "fn" param) that can be passed to sync.invoke
+       */
+      type InvokeAction = 'sync'|'syncRecords'|'listCollisions'|'removeCollision'
+
+      /**
+       * Options that can be passed to sync.invoke
+       */
+      interface InvokeOptions {
+          fn: InvokeAction
+
+          // TODO: we should define this in more detail
+          [key: string]: any
+      }
+
+      /**
+       * Interfaces that describe how sync responses should be structured
+       */
+      namespace HandlerResults {
+          interface Create {
+              uid: string
+              data: Object
+          }
+
+          interface Read {
+              [key: string]: any
+          }
+
+          interface Update {
+              [key: string]: any
+          }
+
+          interface Delete {
+              [key: string]: any
+          }
+
+          interface List {
+              [uid: string]: Object
+          }
+      }
+
+      /**
+       * Unique callback structures for sync handlers.
+       *
+       * Can be used by other modules to create pre-built sync compliant handlers.
+       */
+      namespace HandlerFunctions {
+          type Create = (dataset: string, data: Object, metaData: any, done: StandardCb<HandlerResults.Create>) => void
```

Comment:
Just noticed a slight inconsistency with these handler functions. The first argument for half of them is `dataset` and the other half have `datasetId` but I think they're all the same thing?

Reply:
Good point. They should be consistent.